src/ClientData/include/ClientData/CaptureData.h

```diff
@@ -288,7 +289,7 @@ class CaptureData {
  private:
   orbit_grpc_protos::CaptureStarted capture_started_;
 
-  orbit_client_data::ProcessData process_;
+  std::unique_ptr<orbit_client_data::ProcessData> process_;
```

Comment:
I thought about that. The thing I don't like about it, is that process_ is actually never optional. We always set it in the constructor, but for reasons, we can not initialize it right-away. And as it does not have a default constructor anymore, we need to do that trick.

Reply:
What I'm saying is that you can do that trick with `optional`. As it avoids heap allocation, it should be the default if you don't need anything else from `unique_ptr`. go/totw/187#common-anti-pattern-delayed-initialization: "If delayed initialization is really important and unavoidable, consider using std::optional with its emplace() method."

Reply:
I kinda dislike the reading of "optional process_", as it is indeed not optional. However, I followed your advice and made it optional and added a comment above.

Reply:
I certainly see what you mean. But I think a similar argument could be made again using `unique_ptr`, even if the name is less "strong". So the tip just suggests to pick the one with the simplest behavior.

But remember: The tip strongly suggests to avoid delayed initialization entirely if possible! I think it would be doable here with an immediately invoked lambda that builds a `ProcessInfo`.

Reply:
Indeed, or just a function or static method...